### PR TITLE
[PD-3883]  calendar updates should not clear search

### PR DESCRIPTION
**Pull request by vigowei**
## What's this PR do?
Prevent search term from being cleared when the calendar is updated. 

## How should this be manually tested?
compare search box behavior on QC and this branch, notice how it's affected by selecting new dates from the calendar.

## Any background context you want to provide?
Added an action to set the search term ('partial') for the currently active tab. This search value is then applied on the backend when the user updates the calendar.

### Front-End

- [ ] Works on IE 11 and Edge
- [ ] Works on Mobile
- [ ] Unit tests added or ticket exists to add tests

### CODE REVIEW

**Comment by aljordan82** [comment]
Looks good, but there is minor issue. I think we need to bind the search field value to the stored `partial` in redux, so when the tab gets changed, the search value gets updated.

File: gulp/src/analytics/actions/calendar-actions.js [hunk 1]
```diff
@@ -26,6 +26,8 @@ export function doSetSelectedRange(startDate, endDate) {
       const dimension = activeNavigation.PageLoadData.dimension;
       // Grab multi filter from the state navigation
       const rawMultiFilter = getState().main.navigation[0].PageLoadData.multifilter;
+      // Check if the search box has value
+      const partial = getState().main.navigation[0].partial || '';
```
Comment (aljordan82) [inline gulp/src/analytics/actions/calendar-actions.js]: `partial` default value can be added when the `navigation` item is created in reducers.

Comment (vigowei) [reply gulp/src/analytics/actions/calendar-actions.js]: I think that's `SET_PAGE_DATA`, I've added a default value for it and removed the OR.

**Comment by vigowei** [comment]
@aljordan82 I agree we should be binding partial to tab, right now it's sort of a global state.. I have another branch for this exact issue, will open a new PR once this one is merged.

File: gulp/src/analytics/actions/calendar-actions.js (hunk 1 above)
Comment (vigowei) [reply gulp/src/analytics/actions/calendar-actions.js]: reverted, will address this in the next PR

**Pull request merged**